python/graphstorm/model/edge_decoder.py

```diff
@@ -602,8 +603,15 @@ def calc_test_scores(self, emb, pos_neg_tuple, neg_sample_type, device):
             assert len(neg_score.shape) == 2
             neg_scores.append(neg_score)
 
-        neg_scores = th.cat(neg_scores, dim=-1).detach().cpu()
-        pos_scores = pos_scores.detach().cpu()
+        neg_scores = th.cat(neg_scores, dim=-1).detach()
+        # gloo with cpu will consume less GPU memory
+        neg_scores = neg_scores.cpu() \
+            if is_distributed() and get_backend() == "gloo" \
```

Comment:
Since we are checking is_distributed inside get_backend, we can remove is_distributed.

Reply:
In get_backend, it is an assertion. we can not move is_distributed.